File: api/src/Entity/Progression.php
<?php

namespace App\Entity;

use App\Repository\ProgressionRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ProgressionRepository::class)]
class Progression
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(inversedBy: 'progressions')]
    private ?TypeProgression $TypeProgression = null;

    #[ORM\ManyToOne(inversedBy: 'progressions')]
    private ?Utilisateur $Utilisateur = null;

    #[ORM\ManyToOne(inversedBy: 'progressions')]
    private ?Ressource $Ressource = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTypeProgression(): ?TypeProgression
    {
        return $this->TypeProgression;
    }

    public function setTypeProgression(?TypeProgression $TypeProgression): static
    {
        $this->TypeProgression = $TypeProgression;

        return $this;
    }

    public function getUtilisateur(): ?Utilisateur
    {
        return $this->Utilisateur;
    }

    public function setUtilisateur(?Utilisateur $Utilisateur): static
    {
        $this->Utilisateur = $Utilisateur;

        return $this;
    }

    public function getRessource(): ?Ressource
    {
        return $this->Ressource;
    }

    public function setRessource(?Ressource $Ressource): static
    {
        $this->Ressource = $Ressource;

        return $this;
    }
}

File: mobile/ressources_re_mobile/lib/classes/Progression.dart
// ignore_for_file: file_names

import 'package:ressources_re_mobile/classes/Ressource.dart';
import 'package:ressources_re_mobile/classes/TypeProgression.dart';
import 'package:ressources_re_mobile/classes/Utilisateur.dart';

class Progression {
  int? id;
  TypeProgression? typeProgression;
  Utilisateur? utilisateur;
  Ressource? ressource;

  Progression();

  int? getId() {
    return id;
  }

  TypeProgression? getTypeProgression() {
    return typeProgression;
  }

  void setTypeProgression(TypeProgression? typeProgression) {
    this.typeProgression = typeProgression;
  }

  Utilisateur? getUtilisateur() {
    return utilisateur;
  }

  void setUtilisateur(Utilisateur? utilisateur) {
    this.utilisateur = utilisateur;
  }

  Ressource? getRessource() {
    return ressource;
  }

  void setRessource(Ressource? ressource) {
    this.ressource = ressource;
  }

  Progression.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    typeProgression = TypeProgression.fromJson(json['TypeProgression']) != null ? TypeProgression.fromJson(json['TypeProgression']) : null;
    utilisateur = Utilisateur.fromJson(json['Utilisateur']);
    ressource = Ressource.fromJson(json['Ressource']);
  }
}
